app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBHelper.java

```diff
@@ -4,40 +4,53 @@
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteOpenHelper;
 
-public class TranslationsDBHelper extends SQLiteOpenHelper {
-
-   private static final String DB_NAME = "translations.db";
-   private static final int DB_VERSION = 1;
-
-   public static class TranslationsTable {
-      public static final String TABLE_NAME = "translations";
-      public static final String ID = "id";
-      public static final String NAME = "name";
-      public static final String TRANSLATOR = "translator";
-      public static final String FILENAME = "filename";
-      public static final String URL = "url";
-      public static final String VERSION = "version";
-   }
-
-   private static final String CREATE_TRANSLATIONS_TABLE =
-           "CREATE TABLE " + TranslationsTable.TABLE_NAME + "(" +
-                   TranslationsTable.ID + " integer primary key, " +
-                   TranslationsTable.NAME + " varchar not null, " +
-                   TranslationsTable.TRANSLATOR + " varchar, " +
-                   TranslationsTable.FILENAME + " varchar not null, " +
-                   TranslationsTable.URL + " varchar, " +
-                   TranslationsTable.VERSION + " integer not null default 0);";
-
-   public TranslationsDBHelper(Context context) {
-      super(context, DB_NAME, null, DB_VERSION);
-   }
-
-   @Override
-   public void onCreate(SQLiteDatabase db) {
+import timber.log.Timber;
+
+class TranslationsDBHelper extends SQLiteOpenHelper {
+
+  private static final String DB_NAME = "translations.db";
+  private static final int DB_VERSION = 2;
+  private static final String CREATE_TRANSLATIONS_TABLE =
+      "CREATE TABLE " + TranslationsTable.TABLE_NAME + "(" +
+          TranslationsTable.ID + " integer primary key, " +
+          TranslationsTable.NAME + " varchar not null, " +
+          TranslationsTable.TRANSLATOR + " varchar, " +
+          TranslationsTable.TRANSLATOR_FOREIGN + " varchar, " +
+          TranslationsTable.FILENAME + " varchar not null, " +
+          TranslationsTable.URL + " varchar, " +
+          TranslationsTable.VERSION + " integer not null default 0);";
+
+  TranslationsDBHelper(Context context) {
+    super(context, DB_NAME, null, DB_VERSION);
+  }
+
+  @Override
+  public void onCreate(SQLiteDatabase db) {
+    db.execSQL(CREATE_TRANSLATIONS_TABLE);
+  }
+
+  @Override
+  public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+    if (newVersion <= oldVersion) {
+      Timber.w("Can't downgrade from version %d to version %d", oldVersion, newVersion);
+      return;
+    }
+
+    if (oldVersion < 2) {
+      // a new column is added and columns are re-arranged
+      db.execSQL("DROP TABLE IF EXISTS " + TranslationsTable.TABLE_NAME);
```

Comment:
the downside of this is that it will break translations until the person re-visits the translation screen to re-build the database.

so can we instead of this, either:
alter table add column, and run a query to set the default to translator for now

or

rename the existing table, create the new table, and insert into the new table from the old table, replacing that column with translator.

what do you think?